src/main/java/org/candlepin/async/tasks/CloudAccountOrgSetupJob.java

```diff
@@ -0,0 +1,211 @@
+import org.candlepin.async.JobExecutionException;
+import org.candlepin.model.Owner;
+import org.candlepin.model.OwnerCurator;
+import org.candlepin.service.CloudProvider;
+import org.candlepin.service.CloudRegistrationAdapter;
+import org.candlepin.service.exception.CloudAccountOrgSetupException;
+import org.candlepin.service.model.CloudAccountData;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+
+import javax.inject.Inject;
+
+public class CloudAccountOrgSetupJob implements AsyncJob {
+
+    private static final Logger log = LoggerFactory.getLogger(CloudAccountOrgSetupJob.class);
+
+    public static final String JOB_KEY = "CloudAccountOrgSetupJob";
+    public static final String JOB_NAME = "Cloud Account Org Setup Job";
+
+    protected static final String CLOUD_ACCOUNT_ID = "cloud_account_id";
+    protected static final String OFFERING_ID = "offering_id";
+    protected static final String CLOUD_PROVIDER = "cloud_provider";
+
+    private final CloudRegistrationAdapter cloudAdapter;
+    private OwnerCurator ownerCurator;
+
+    @Inject
+    public CloudAccountOrgSetupJob(CloudRegistrationAdapter cloudAdapter, OwnerCurator ownerCurator) {
+
+        this.cloudAdapter = Objects.requireNonNull(cloudAdapter);
+        this.ownerCurator = Objects.requireNonNull(ownerCurator);
+    }
+
+    /**
+     * Executes this job.
+     *
+     * @param context
+     *     the execution context for the job, containing the arguments and other context-specific data
+     * @throws JobExecutionException
+     *     if an exception occurs during job execution
+     */
+    @Override
+    public void execute(JobExecutionContext context) throws JobExecutionException {
+        JobArguments args = context.getJobArguments();
+
+        String accountId = args.getAsString(CLOUD_ACCOUNT_ID);
+        String offeringId = args.getAsString(OFFERING_ID);
+        CloudProvider cloudProviderShortName = args.getAs(CLOUD_PROVIDER, CloudProvider.class);
+
+        try {
+            CloudAccountData accountData = this.cloudAdapter.setupCloudAccountOrg(
+                accountId, offeringId, cloudProviderShortName);
+
+            Owner owner = ownerCurator.lockAndLoadByKey(accountData.ownerKey());
+            if (owner == null) {
+                if (accountData.isAnonymous() == null) {
+                    throw new IllegalStateException("Owner does not exist in candlepin");
+                }
+                else {
+                    Owner newOwner = new Owner()
+                        .setKey(accountData.ownerKey())
+                        .setAnonymous(accountData.isAnonymous())
+                        .setClaimed(false);
+                    ownerCurator.create(newOwner);
+                }
+            }
+            else {
+                owner.setAnonymous(accountData.isAnonymous()).setClaimed(false);
+                ownerCurator.saveOrUpdate(owner);
+            }
```

Comment:
If the owner already exists in candlepin, we shouldn't be updating it. let's remove this section

Reply:
done